src/Design/03_InitialPoC_trans.purs

```diff
@@ -1,56 +1,52 @@
 module PSPS.Design.InitialPoC.Trans where
 
-import PSPS (class Bind, class Eval, App, kind Expr, kind ModuleName)
+import FFI.PSPS.Design.InitialPoC.Trans (class FFI_PSPS_Design_InitialPoC_Trans_Eval, Foreign'intAdd, Foreign'intMul)
+import PSPS (class Bind, class Eval, Abs, App, Foreign, kind Expr, kind ModuleName)
+import Type.Data.Peano (Succ, Z)
 
 foreign import data PSPS_Design_InitialPoC_Trans :: ModuleName
 
 class PSPS_Design_InitialPoC_Trans_Eval (expr :: Expr) (result :: Expr) | expr -> result
 
 class PSPS_Design_InitialPoC_Trans_Bind (path :: Symbol) (expr :: Expr) | path -> expr
 
-foreign import kind Maybe
-
-foreign import data Just :: Expr -> Maybe
-
-foreign import data Nothing :: Maybe
-
-foreign import data Maybe0 :: Maybe -> Expr
-
-foreign import data Maybe1 :: (Expr -> Maybe) -> Expr
-
-instance iPSPS_Design_InitialPoC_Trans_Bind'Just :: PSPS_Design_InitialPoC_Trans_Bind "Just" (Maybe1 Just)
-
-instance iPSPS_Design_InitialPoC_Trans_Bind'Nothing :: PSPS_Design_InitialPoC_Trans_Bind "Nothing" (Maybe0 Nothing)
-
-instance iBind'Just :: PSPS_Design_InitialPoC_Trans_Bind "Just" o => Bind PSPS_Design_InitialPoC_Trans "Just" o
+instance iPSPS_Design_InitialPoC_Trans_Bind'intAdd :: PSPS_Design_InitialPoC_Trans_Bind "intAdd" (Abs "intAdd" (Succ Z) (Abs "intAdd" Z (Foreign Foreign'intAdd)))
 
-instance iBind'Nothing :: PSPS_Design_InitialPoC_Trans_Bind "Nothing" o => Bind PSPS_Design_InitialPoC_Trans "Nothing" o
+instance iPSPS_Design_InitialPoC_Trans_Eval'intAdd'intAdd1'App ::
```

Comment:
As promised, in `Eval`, we create an application of the abs to `a`. Remember that the way to read this is that "if you give me what's on the left, I, compiler, will give you what's on the right." So we say "Ok compiler, here's the definition of `intAdd` fetched from `Bind` above (`(Abs "intAdd" (Succ Z) (Abs "intAdd" Z (Foreign Foreign'intAdd)))`) and I want to apply it to `a`". The compiler says "No prob, I'll evaluate `a` to `a'` and return to you a curried function on the inside of which is the foreign function applied to `a'`.